src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -1873,10 +1898,149 @@ public void testEventStreamPolling() throws Exception {
     assertNotNull(eventList);
     assertThat(eventList, not(empty()));
 
-    final Event pullEvent = eventList.get(0);
-    final Event createEvent = eventList.get(1);
-    final Event startEvent = eventList.get(2);
-    eventStreamAssertions(pullEvent, createEvent, startEvent, container.id(), BUSYBOX_LATEST);
+    imagePullEventAssertions(eventList.get(0));
+
+    // create and start event assertions
+    containerEventAssertions(eventList.get(1), containerId, containerName,
+            "create", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(2), containerId, containerName,
+            "start", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(3), containerId, containerName,
+            "die", BUSYBOX_LATEST);
+    containerEventAssertions(eventList.get(4), containerId, containerName,
+            "destroy", BUSYBOX_LATEST);
+  }
+
+  @Test(timeout = 10000)
+  public void testEventTypes() throws Exception {
+    requireDockerApiVersionAtLeast("1.22", "Event types");
+
+    final String volumeName = randomName();
+    final String containerName = randomName();
+    final String mountPath = "/anywhere";
+    final Volume volume = Volume.builder().name(volumeName).build();
+    final HostConfig hostConfig = HostConfig.builder()
+            .binds(Bind.from(volume).to(mountPath).build())
+            .build();
+    final ContainerConfig config = ContainerConfig.builder()
+            .image(BUSYBOX_LATEST)
+            .hostConfig(hostConfig)
+            .build();
+
+    // Wait once to clean our event "palette" of events from other tests
+    Thread.sleep(1000);
```

Comment:
This is probably fine for now, but is there a way to check there are no leftover events? Perhaps we can use [awaitility](https://github.com/awaitility/awaitility) in the future here.